nixos/modules/services/monitoring/telegraf.nix

```diff
@@ -0,0 +1,76 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.telegraf;
+
+  configFile = pkgs.runCommand "config.toml" {
+    buildInputs = [ pkgs.remarshal ];
+  } ''
+    remarshal -if json -of toml \
+      < ${pkgs.writeText "config.json" (builtins.toJSON cfg.extraConfig)} \
+      > $out
+  '';
+in
+{
+
+  ###### interface
+
+  options = {
+
+    services.telegraf = {
+
+      enable = mkOption {
+        default = false;
+        description = "Whether to enable the telegraf server";
+        type = types.bool;
+      };
+
+      package = mkOption {
```

Comment:
Package overrides is not really about `nix-env`, it is a general mechanism for extending/changing the package set.

For example,

```
  nixpkgs.config.packageOverrides = super: {
     telegraf = super.callPackage ./my/newer/telegraf.nix {};
  }
```

ought to affect uses of `pkgs.telegraf` in the service definition. You should be able to do the same via `~/.nixpkgs/config.nix`.